File: code_eval/java/easy/Challenge21.java
package easy;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;

/**
 * Challenge 21 : Sum integers
 * Created by cschalk on 3/21/16.
 */
public class Challenge21 {

    public static void main (String[] args) throws Exception {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            sumTheDigits(line);
        }
    }

    private static void sumTheDigits(String input) {
        int sum = 0;
        for (char c : input.toCharArray()) {
            int digit = Integer.parseInt("" + c);
            sum += digit;
        }
        System.out.println(sum);
    }

}

File: code_eval/cpp/easy/challenge_21.cpp
// Created by cschalk on 3/20/16.
// Challenge 21 : Sum of digits in a number
#include <fstream>
#include <iostream>

using namespace std;

int sumOfDigits(int in) {
    int sum = 0;
    while ( in > 0 ) {
        sum += in % 10;
        in /= 10;
    }
    return sum;
}

int main(int argc, char ** args) {
    std::ifstream input {args[1]};
    int number;
    while(input >> number) {
        std::cout << sumOfDigits(number) << std::endl;
    }

    return 0;

}


